components/core/src/clp/BufferedFileReader.cpp

```diff
@@ -22,102 +30,47 @@ namespace {
  * @return ErrorCode_EndOfFile on EOF
  * @return ErrorCode_Success on success
  */
-auto read_into_buffer(int fd, char* buf, size_t num_bytes_to_read, size_t& num_bytes_read)
-        -> ErrorCode;
-
-auto read_into_buffer(int fd, char* buf, size_t num_bytes_to_read, size_t& num_bytes_read)
-        -> ErrorCode {
+auto read_into_buffer(
+        ReaderInterface* reader,
+        char* buf,
+        size_t num_bytes_to_read,
+        size_t& num_bytes_read
+) -> ErrorCode;
+
+auto read_into_buffer(
+        ReaderInterface* reader,
+        char* buf,
+        size_t num_bytes_to_read,
+        size_t& num_bytes_read
+) -> ErrorCode {
     num_bytes_read = 0;
-    while (true) {
-        auto const bytes_read = ::read(fd, buf, num_bytes_to_read);
-        if (0 == bytes_read) {
-            break;
-        }
-        if (bytes_read < 0) {
-            return ErrorCode_errno;
-        }
-
-        buf += bytes_read;
-        num_bytes_read += bytes_read;
-        num_bytes_to_read -= bytes_read;
-        if (num_bytes_read == num_bytes_to_read) {
-            return ErrorCode_Success;
-        }
-    }
+    auto const error_code = reader->try_read(buf, num_bytes_to_read, num_bytes_read);
     if (0 == num_bytes_read) {
         return ErrorCode_EndOfFile;
     }
-    return ErrorCode_Success;
+    return error_code;
```

Comment:
"If try_read doesn't read up to the num_bytes_to_read, it should return ErrorCode_EndOfFile"
Are you sure? My understanding is that as long as non-zero bytes are read, it will return success. ErrorCode_EndOfFile is only returned if the interface doesn't contain any data

That said, it doesn't change your conclusion on this line, but it does affect another comment you made

Reply:
Sorry, my mistake, you're right.